src/main/java/com/pinterest/secor/monitoring/MicroMeterMetricCollector.java

```diff
@@ -63,13 +78,21 @@ public void increment(String label, int delta, String topic) {
 
     @Override
     public void metric(String label, double value, String topic) {
-        Metrics.gauge(label, Collections.singletonList(
-            Tag.of("topic", topic)), new AtomicDouble(0)).set(value);
+        gauge(label, value, topic);
     }
 
     @Override
     public void gauge(String label, double value, String topic) {
+        String key = label + "_" + topic;
+        if (mGaugeCache.size() <= mConfig.getMicroMeterCacheSize()) {
```

Comment:
I think we should only do the cache size check and issue error message when the cache needs to grow, not each time gauge is called (there will be too frequent)